### ISSUE TITLE
Indexed Monad

### ISSUE THREAD

**Issue by gaku-sei** [Risto-Stevcev/bs-abstract#18]
Is there any plan to bring the Indexed (Monads) to `bs-abstract`? It could be especially useful when state management is involved.

I made a small gist [here](https://gist.github.com/gaku-sei/434d4de22dd66402728542cb9c3c7d35). I tried to follow the same naming convention as the repository.

Please let me know if you're interested 😃

**Comment by Risto-Stevcev** [comment]
Yeah definitely! I mostly use GADTs or poly variants for that kind of stuff but it's useful to have in the library for sure
